backend/src/api/billing/billing.api.ts

```diff
@@ -0,0 +1,55 @@
+import { FastifyPluginAsync, FastifyRequest } from 'fastify';
+
+import { BillingApiPath, HttpCode, HttpMethod } from '~/common/enums/enums';
+import { BillingReplenishParamsDto } from '~/common/types/types';
+import {
+  billing as billingService,
+  user as userService,
+  userDetails as userDetailsService,
+} from '~/services/services';
+import { billingReplenishParams as billingReplenishParamsValidationSchema } from '~/validation-schemas/validation-schemas';
+
+type Options = {
+  services: {
+    billing: typeof billingService;
+    user: typeof userService;
+    userDetails: typeof userDetailsService;
+  };
+};
+
+const initBillingApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
+  const { billing: billingService } = opts.services;
+
+  fastify.route({
+    method: HttpMethod.POST,
```

Comment:
Not sure about the POST method here, because the payment is handled by Stripe and you only update the balance. What about PATCH?

Reply:
I chose POST because of the payment method (as Stripe hanldes it with POST). I think, we need coaches adivice here

Reply:
Use PATCH here